packages/api/login.go

```diff
@@ -153,24 +153,23 @@ func login(w http.ResponseWriter, r *http.Request, data *apiData, logger *log.En
 	if r, ok := data.params["role_id"]; ok {
 		role := r.(int64)
 		if role > 0 {
-			ok, err := model.MemberHasRole(nil, ecosystemID, wallet, role)
-			if err != nil {
-				logger.WithFields(log.Fields{
+			getRoleErrorFields := func() log.Fields {
```

Comment:
you could do requestLogger := logger.WithFields(log.Fields{...}} instead of that